src/mod_push.erl

```diff
@@ -432,6 +439,38 @@ notify(LUser, LServer, Clients) ->
 	      notify(LServer, PushLJID, Node, XData, HandleResponse)
       end, Clients).
 
+-spec notify(binary(), binary(), [push_session()], message()) -> ok.
+notify(LUser, LServer, Clients, #message{to = #jid{luser = ToUser, lserver = ToServer}, from = #jid{luser = FromUser, lserver = FromServer}, body = [#text{data = BodyData}]} = Pkt) ->
+    lists:foreach(
+      fun({TS, PushLJID, Node, XData}) ->
+              [Host] = ?MYHOSTS,
+              AccessKey = gen_mod:get_module_opt(Host, ?MODULE, access_key),
+              AccessSecret = gen_mod:get_module_opt(Host, ?MODULE, access_secret),
+              PushMsgParams = epushmsg:new_params(AccessSecret, AccessKey),
+              case XData of
+                  #xdata{fields = XDataFields} ->
+                      lists:foreach(
+                        fun(XDataField) ->
+                                case XDataField of
+                                    #xdata_field{var = <<"secret">>, values = [Secret]} ->
+                                        Audience = epushmsg:new_namedUser(Secret),
+                                        Alert = epushmsg:new_alert(<<"New Message">>),
+                                        Payload = epushmsg:new_payload(Audience, Alert, all),
+
+                                        PushMsgParams2 = PushMsgParams#pushmsg_params{payload=Payload},
+                                        StatusCode = epushmsg:push(PushMsgParams2),
```

Comment:
Is there anyway to log or handle if something wrong happened when it push notification ?

Reply:
if something wrong happened it will just ignored (not result in crash) as with nature of erlang system. we might need to log it though